listener/processor.py

```diff
@@ -77,3 +78,68 @@ async def _db_account_system_lookup(self, conn, org_id: str, inventory_id: str)
             if result["inserted"]:
                 NEW_RH_ACCOUNT.inc()
             return result["id"], None
+
+    async def _db_operating_system_lookup(self, conn, operating_system: dict) -> int:
+        """Lookup internal DB id for operating system"""
+        os_name = operating_system.get("name", "UNKNOWN")
+        os_major = operating_system.get("major", 0)
+        os_minor = operating_system.get("minor", 0)
```

Comment:
I think we should fail when these keys are not in operating_system dict because otherwise we will insert UNKNOWN-0.0 into DB